### ISSUE TITLE
SSL passthrough not working in 1.7.4

### ISSUE THREAD

**Issue by sivanov-nuodb** [haproxytech/kubernetes-ingress#416]
**What happened?**

Ingress resources configured with `ssl-passthrough=true` don't work properly in v1.7.4 . 
The exact same Ingress resource works as expected in v1.6.7.

The HAProxy doesn't route SSL traffic with the correct SNI hostname to the backend Kubernetes service. There is no logging or anything else indicating that the traffic is dropped on the floor.

**What did you expect to happen?**

Ingress with `ssl-passthrough=true` should instruct HAProxy to handle the incoming SSL traffic with the correct SNI hostname via the `tcp` frontend and route it to the corresponding backend based on SNI mapping without offloading the SSL handshake.

**How can we reproduce it (as minimally and precisely as possible)?**

_Steps to reproduce:_

Install HAProxy Ingress with the following values file 
[haproxy.txt](https://github.com/haproxytech/kubernetes-ingress/files/7950246/haproxy.txt).

```
helm install haproxy-ingress haproxytech/kubernetes-ingress -n haproxy-system -f haproxy.yaml
```

Create HTTP server Deployment capable of speaking HTTPS in namespace _foo_ which exposes port 8888 and create Service _foo-svc_ for its Pods.
Create the following Ingress resource:

```yaml
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: foo-ingress
  namespace: foo
  annotations:
    kubernetes.io/ingress.class: "haproxy"
    ingress.kubernetes.io/ssl-passthrough: "true"
    haproxy.org/ssl-passthrough: "true"
spec:
  rules:
  - host: foo.example.com
    http:
      paths:
      - path: /
        pathType: "Prefix"
        backend:
          service:
            name: foo-svc
            port:
              number: 8888
```

Create DNS record for `foo.example.com` that points to the hostname or IP address of the HAProxy LoadBlancer service external IP.  

Try to access the service using _curl_:

```
curl -k --cert foo.pem  https://foo.example.com/api/1/version -vvv
*   Trying 34.132.222.14...
* TCP_NODELAY set
* Connected to foo.example.com (34.132.222.14) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to foo.example.com:443
* Closing connection 0
curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to foo.example.com:443
```

_Controller logs_:

```
2022/01/27 08:50:58 HAProxy Ingress Controller v1.7.4 ad1d939
2022/01/27 08:50:58 Build from: https://github.com/haproxytech/kubernetes-ingress
2022/01/27 08:50:58 Build date: 2022-01-13T16:46:24
2022/01/27 08:50:58 ConfigMap: haproxy-system/haproxy-ingress-kubernetes-ingress
2022/01/27 08:50:58 Ingress class: haproxy
2022/01/27 08:50:58 Empty Ingress class: false
2022/01/27 08:50:58 Publish service: haproxy-system/haproxy-ingress-kubernetes-ingress
2022/01/27 08:50:58 Default backend service: haproxy-system/haproxy-ingress-kubernetes-ingress-default-backend
2022/01/27 08:50:58 Default ssl certificate: haproxy-system/haproxy-ingress-kubernetes-ingress-default-cert
2022/01/27 08:50:58 Frontend HTTP listening on: 0.0.0.0:80
2022/01/27 08:50:58 Frontend HTTPS listening on: 0.0.0.0:443
2022/01/27 08:50:58 DEBUG   Kubernetes Informers resync period: 10m0s
2022/01/27 08:50:58 Controller sync period: 5s
...
2022/01/27 08:56:01 DEBUG   service/service.go:130 Service 'foo/foo-svc': new backend 'foo-foo-svc-8888-tcp', reload required
2022/01/27 08:56:01 DEBUG   handler/https.go:178 SSLPassthrough enabled, reload required
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_SET_VAR' created, reload required
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_SET_VAR' created, reload required
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_SET_VAR' created, reload required
2022/01/27 08:56:01 ERROR   rules/main.go:199 failed to create a REQ_INSPECT_DELAY rule: 14: unsupported action in tcp_request_rule
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_ACCEPT_CONTENT' created, reload required
2022/01/27 08:56:01 DEBUG   maps/main.go:124 Map file 'sni' updated, reload required
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_SET_VAR' created, reload required
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_SET_VAR' created, reload required
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_SET_VAR' created, reload required
2022/01/27 08:56:01 ERROR   rules/main.go:199 failed to create a REQ_INSPECT_DELAY rule: 14: unsupported action in tcp_request_rule
2022/01/27 08:56:01 DEBUG   rules/main.go:204 New HAProxy rule 'REQ_ACCEPT_CONTENT' created, reload required
2022/01/27 08:56:01 INFO    controller.go:211 HAProxy reloaded
[WARNING]  (251) : Reexecuting Master process
[WARNING]  (251) : config: Can't get version of the global server state file '/var/state/haproxy/global'.
[WARNING]  (273) : Proxy healthz stopped (cumulated conns: FE: 2, BE: 0).
Proxy healthz stopped (cumulated conns: FE: 2, BE: 0).
[WARNING]  (273) : Proxy http stopped (cumulated conns: FE: 0, BE: 0).
Proxy http stopped (cumulated conns: FE: 0, BE: 0).
[WARNING]  (273) : Proxy https stopped (cumulated conns: FE: 0, BE: 0).
Proxy https stopped (cumulated conns: FE: 0, BE: 0).
[WARNING]  (273) : Proxy stats stopped (cumulated conns: FE: 0, BE: 0).
Proxy stats stopped (cumulated conns: FE: 0, BE: 0).
[WARNING]  (273) : Stopping frontend GLOBAL in 0 ms.
[WARNING]  (273) : Stopping backend haproxy-system-haproxy-ingress-kubernetes-ingress-default-backend-http in 0 ms.
Stopping backend haproxy-system-haproxy-ingress-kubernetes-ingress-default-backend-http in 0 ms.
[NOTICE]   (251) : New worker #1 (279) forked
[WARNING]  (251) : Former worker #1 (273) exited with code 0 (Exit)
```

_HAPproxy config_:

[haproxy-conf.txt](https://github.com/haproxytech/kubernetes-ingress/files/7950322/haproxy-conf.txt)

**HAProxy Ingress version**

`1.7.4`

Note: The exact same scenario work with v1.6.7.

**Kubernetes version**

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"ca643a4d1f7bfe34773c74f79527be4afd95bf39", GitTreeState:"clean", BuildDate:"2021-07-15T21:04:39Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5-gke.1302", GitCommit:"639f3a74abf258418493e9b75f2f98a08da29733", GitTreeState:"clean", BuildDate:"2021-10-21T21:35:48Z", GoVersion:"go1.16.7b7", Compiler:"gc", Platform:"linux/amd64"}
```

**Cloud provider**

GKE

**Container runtime (CRI) and and version (if applicable)**

N/A

**Comment by Mo3m3n** [comment]
Hi 
A workaround is available here https://github.com/haproxytech/kubernetes-ingress/issues/414#issuecomment-1018747883
And fix is already on master, it will be backported for next `v1.7`.

**Comment by sivanov-nuodb** [comment]
@Mo3m3n Thanks for the quick response!
Just tested and the workaround doesn't work for me.

**Comment by Mo3m3n** [comment]
Can you give the `haproxytech/kubernetes-ingress:nightly` a try where this issue is supposed to be fixed ?

**Comment by sivanov-nuodb** [comment]
I can confirm that it works with `nightly` tag.

**Comment by Mo3m3n** [comment]
Then the workaround referenced above should be working too.
You need to set it in the configmap and to be sure it is applied you should see `timeout client` directive in the defaults section of the generated haproxy.cfg

**Comment by Mo3m3n** [comment]
Fix available in `v1.7.5`
